packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1512,6 +1519,58 @@ export class ContainerRuntime
 	 * The options to apply to this ContainerRuntime instance (including internal options hidden from the public API)
 	 */
 	private readonly runtimeOptions: Readonly<Required<IContainerRuntimeOptionsInternal>>;
+	public readonly pkgVersion = pkgVersion;
+	/**
+	 * The current generation of the Runtime layer. This is used to ensure compatibility between the Runtime and
+	 * other layers.
+	 */
+	private readonly generation = currentRuntimeGeneration;
+	/**
+	 * A list of features that the Loader layer is required to support to be compatible with this Runtime.
+	 */
+	private readonly requiredFeaturesFromLoader: string[] = [];
+	/**
+	 * A list of features supported by the Runtime layer. This is exposed to the Loader layer which uses
+	 * it to determine if the Runtime is compatible with the Loader.
+	 */
+	private readonly supportedFeaturesForLoader: ReadonlyMap<string, unknown> = new Map([
+		/* This is the minimum generation of the Loader this Runtime supports. */
+		["minSupportedGeneration", 1],
+	]);
+	public get supportedFeatures(): ReadonlyMap<string, unknown> {
+		return this.supportedFeaturesForLoader;
+	}
+	private validateLoaderCompatibility(
+		loaderSupportedFeatures?: ReadonlyMap<string, unknown>,
+		loaderVersion?: string,
+	): void {
+		if (loaderSupportedFeatures === undefined) {
```

Comment:
Same as on Loader side - I don't think this belongs, but maybe I'm missing something

Reply:
I'm hoping we can remove it, so that this `validate` helper can have no logic at all besides calling `checkLayerCompatibility` and throwing.  The more concentrated we can have this logic, the better.

Reply:
Updated this.